modules-js/react-fleet/src/components/Header.tsx

```diff
@@ -21,22 +22,19 @@ export default function Header(
 
   useEffect(() => {
     let lastPos: number = 0;
-    let isScrolling: boolean = false;
+    let timer;
 
     const handleScroll = (): void => {
       lastPos = window.scrollY;
 
-      if (!isScrolling) {
-        window.requestAnimationFrame(() => {
-          setScrollPosition(lastPos);
-
-          handleScrollStop();
-
-          isScrolling = false;
-        });
-
-        isScrolling = true;
+      if (timer) {
+        // Update scroll position while user is scrolling
+        window.requestAnimationFrame(() => setScrollPosition(lastPos));
```

Comment:
I think you don't need to keep `scrollPosition` around anymore, right? You can just check `window.scrollY` inside of `handleScrollStop`. Should prevent some re-renderings if you're not changing state.

Reply:
hah, you’re right - I was being myopic.